activerecord/lib/active_record/connection_adapters/postgresql/schema_statements.rb

```diff
@@ -447,15 +447,22 @@ def rename_column(table_name, column_name, new_column_name) # :nodoc:
         end
 
         def add_index(table_name, column_name, **options) # :nodoc:
-          index, algorithm, if_not_exists = add_index_options(table_name, column_name, **options)
-
-          create_index = CreateIndexDefinition.new(index, algorithm, if_not_exists)
-          result = execute schema_creation.accept(create_index)
+          create_index = build_create_index_definition(table_name, column_name, **options)
+          result = execute(create_index.ddl)
 
+          index = create_index.index
           execute "COMMENT ON INDEX #{quote_column_name(index.name)} IS #{quote(index.comment)}" if index.comment
           result
         end
 
+        def build_create_index_definition(table_name, column_name, **options)
```

Comment:
Should apps consume this in the ExecutionStrategy? If so it should be documented, if not it should be `# :nodoc:`.

Reply:
I'm feeling a bit torn on this... it's valuable API for a subset of advanced API consumers, but I'm also conscious that this puts it in the middle of an API doc page that's very ordinary-user-facing.

I wonder if we should pull these sorts of methods out into a module just for the sake of the physical separation in the docs... or even `:nodoc:` them [for now?] while still providing descriptive documentation comments. 🤔

Reply:
Hey @matthewd , that's a great point! To have them mixed in with other migration methods is probably confusing given that the callee in the consuming app is different (a migration vs a migration strategy, with the latter being used only by a subset of applications).

I can extract them into a separate module 👍 I'm happy to also `:nodoc:` them, although maybe having the physical separation as you mentioned is enough to clarify their intent.

Reply:
Let's `nodoc` them (but leave the documentation). We can move and reveal them later when the API is more settled.

Reply:
Hrm ok I commented this before I saw the separate module PR. I am very torn on this. 😅 

It's much easier to make a private API that's no-doc'd public than it is to un-public a module/class. My thinking is we no-doc for now with the intention of moving them later to make them public. Once it's clear what needs to be separated the next step could be making a module that is designed to inform users how to consume these APIs for more complex migrations.